Makefile

```diff
@@ -171,6 +182,16 @@ checkvariantchange:
 		fi \
 	fi
 
+# Create the library weak_stubs_libs
+$(WEAK_STUBS_LIB): $(WEAK_STUBS_OBJ)
+	@${ECHOE} ${GREEN}Building weak_stubs_libs...${NC}
+	@$(AR) rcs $@ $^
+
+# Rule to compile .c files into .o files in the correct directory
+$(BUILD_DIR)/skelton/src/%.o: $(BUILD_WEAK_STUBS_SRC)/%.c
```

Comment:
think this seems wrong

WEAK_STUBS_OUTPUT_DIR=$(BUILD_DIR)/weak_stubs/obj:

$(WEAK_STUBS_OUTPUT_DIR)/%.o: $(BUILD_WEAK_STUBS_SRC)/%.c

should it be something like this?